presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java

```diff
@@ -1392,6 +1393,106 @@ public void testMaterializedViewOptimizationWithDerivedFields()
+        String view = "lineitem_partitioned_view_derived_fields";
+
+        try {
+            queryRunner.execute(format("CREATE TABLE %s WITH (partitioned_by = ARRAY['ds', 'shipmode']) AS " +
+                    "SELECT discount, extendedprice, '2020-01-01' as ds, shipmode FROM lineitem WHERE orderkey < 1000 " +
+                    "UNION ALL " +
+                    "SELECT discount, extendedprice, '2020-01-02' as ds, shipmode FROM lineitem WHERE orderkey > 1000", baseTable));
+
+            assertUpdate(format(
+                    "CREATE MATERIALIZED VIEW %s WITH (partitioned_by = ARRAY['mvds', 'shipmode']) AS " +
+                            "SELECT discount as mv_discount , extendedprice as mv_extendedprice , ds as mvds, shipmode FROM %s", view, baseTable));
+
+            assertTrue(getQueryRunner().tableExists(getSession(), view));
+            assertUpdate(format("INSERT INTO %s(mv_discount , mv_extendedprice , mvds, shipmode) " +
+                            "select discount, extendedprice, ds, shipmode from %s where ds='2020-01-01'",
+                    view, baseTable), 1004);
+
+            String baseQuery = format("SELECT sum(discount + extendedprice * discount) as random from %s group by ds, shipmode", baseTable);
+
+            queryRunner.execute(queryOptimizationWithMaterializedView, baseQuery);
+        }
+        finally {
+            queryRunner.execute("DROP TABLE IF EXISTS " + view);
+            queryRunner.execute("DROP TABLE IF EXISTS " + baseTable);
+        }
+    }
+
+    @Test(enabled = false)
+    public void testBaseToViewConversionWithWhereCondition()
+    {
+        Session queryOptimizationWithMaterializedView = Session.builder(getSession())
+                .setSystemProperty(QUERY_OPTIMIZATION_WITH_MATERIALIZED_VIEW_ENABLED, "true")
+                .build();
+        QueryRunner queryRunner = getQueryRunner();
+        try {
+            queryRunner.execute("CREATE TABLE orders_partitioned WITH (partitioned_by = ARRAY['ds']) AS " +
+                    "SELECT orderkey, orderpriority, '2020-01-01' as ds FROM orders WHERE orderkey < 1000 " +
+                    "UNION ALL " +
+                    "SELECT orderkey, orderpriority, '2019-01-02' as ds FROM orders WHERE orderkey > 1000");
+
+            assertUpdate("CREATE MATERIALIZED VIEW test_orders_view WITH (partitioned_by = ARRAY['ds']) " +
+                    "AS SELECT orderkey as ok, orderpriority, ds FROM orders_partitioned");
+            assertTrue(getQueryRunner().tableExists(getSession(), "test_orders_view"));
+            assertUpdate("INSERT INTO test_orders_view(ok, orderpriority, ds) " +
+                    "select orderkey, orderpriority, ds from orders_partitioned where ds='2020-01-01'", 255);
+
+            String baseQuery = "SELECT orderkey from orders_partitioned where orderkey <  10000 and orderkey > 500";
+
+            queryRunner.execute(queryOptimizationWithMaterializedView, baseQuery);
```

Comment:
Given these are unit tests, what's your plan for the ASSERTs? We don't need them now, but just think over it. You need to figure them out before merge